src/laser_geometry.cpp

```diff
@@ -43,10 +43,20 @@ namespace laser_geometry
 
     // Fill the ranges matrix
     for (unsigned int index = 0; index < scan_in.ranges.size(); index++)
+    {
+      if (preservative)
+      {
+        ranges (0, index) = (double) (scan_in.ranges[index] == std::numeric_limits<float>::infinity()
+                          ? scan_in.range_max : scan_in.ranges[index]);
+        ranges (1, index) = (double) (scan_in.ranges[index] == std::numeric_limits<float>::infinity()
```

Comment:
Only the `-Inf` could make some sense, since according the REP-117 it means there's an obstacle somewhere below `range_min`. But I don't think we need that for now.

:+1: